dapps/shop/backend/routes/affiliate.js

```diff
@@ -24,6 +27,68 @@ function authAffiliate(req, res, next) {
   next()
 }
 
+/**
+ * Formats a BigQuery product row into a dshop product obj
+ */
+function bqProductFormatter(product) {
+  const listingIdFromProductId = (id) => {
+    const parts = id.split('-')
+    parts.pop()
+    return parts.join('-')
+  }
+  const makeId = (product) => {
+    // product ID is only in the IPFS path returned
+    const productId = product.ipfs_path.split('/')[2]
+    // listing ID is part of the "product_id" returned by BQ
+    const listingId = listingIdFromProductId(product.product_id)
+    return `${listingId}-${productId}`
+  }
+
+  const id = makeId(product)
+  const {
+    price,
+    title,
+    image
+  } = product
+
+  return {
+    id,
+    data: `/ips/${product.ipfs_path}`,
+    price,
+    title,
+    image
+  }
+}
+
+/**
+ * Get the products from BigQuery for a specific shop
+ */
+async function fetchAffiliateProducts(listingId) {
```

Comment:
I think we were going to have the 50 most recent products from any store, instead of a specific store? Maybe we can add that as an option?

Reply:
Did not know it was not going to be shop specific.  Perhaps only limit it to a specific shop if an auth token is provided?

Reply:
Also, most recent 50?  I don't really have any idea about the use case of this endpoint, but that seems... unexpected.  It'll almost always be from one or two stores, and recency doesn't mean much when it comes to affiliates?

Reply:
Made filtering by shop optional.  If no auth token is provided, it won't filter on listing ID.

Reply:
limited to 50.